openslides_backend/shared/mixins/user_scope_mixin.py

```diff
@@ -160,7 +176,196 @@ def check_permissions_for_scope(
                 self.datastore,
                 self.user_id,
                 CommitteeManagementLevel.CAN_MANAGE,
-                committees,
+                [ci for ci in committees_to_meetings.keys()],
             ):
                 return
-            raise MissingPermission({OrganizationManagementLevel.CAN_MANAGE_USERS: 1})
+            meeting_ids = {
+                meeting_id
+                for mids in committees_to_meetings.values()
+                for meeting_id in mids
+            }
+            if not meeting_ids or not self.check_for_admin_in_all_meetings(
+                instance_id, meeting_ids
+            ):
+                raise MissingPermission(
+                    {
+                        OrganizationManagementLevel.CAN_MANAGE_USERS: 1,
+                        **{
+                            Permissions.User.CAN_UPDATE: meeting_id
+                            for meeting_id in meeting_ids
+                        },
+                    }
+                )
+
+    def check_for_admin_in_all_meetings(
+        self,
+        instance_id: int,
+        b_meeting_ids: set[int] | None = None,
+    ) -> bool:
+        """
+        This is used during a permission check for scope request, user.update/create with payload fields A and F and during
+        user altering actions like user.delete or set_default_password.
+        Checks if the requesting user has permissions to manage participants in all of requested users meetings
+        but requested user doesn't have any of these permissions. See backend issue 2522 on github for more details.
+        Also checks requested user has no committee management rights if a dict was provided instead of an id.
+        An ID will be provided during user deletion.
+        Returns true if permissions are given. False if not. Raises no Exceptions.
+        """
+        if not self._check_not_committee_manager(instance_id):
+            return False
+
+        if not (
+            intersection_meetings := self._collect_intersected_meetings(b_meeting_ids)
+        ):
+            return False
+        assert isinstance(intersection_meetings, dict)
+        admin_meeting_users = self._collect_admin_meeting_users(intersection_meetings)
+        meeting_to_admin_users = self._collect_admin_users(admin_meeting_users)
+        return self._analyze_meetings(meeting_to_admin_users, instance_id)
```

Comment:
```suggestion
        return self._analyze_meetings(meeting_to_admin_users, instance_id)
```
`_analyze_meetings` has very admin-related internal variable names, but no actual admin-specific logic. Moving the calculation of `meeting_to_admin_users` into `_analyze_meetings` would be more sensible.